### PR TITLE
chore: CE companion to pkg export

### PR DESCRIPTION
**Pull request by nidhi-nair**
Companion to https://github.com/appsmithorg/appsmith-ee/pull/3520

## Summary by CodeRabbit

- **Refactor**
	- Enhanced export functionalities for applications and actions.
- **New Features**
	- Advanced export capabilities for applications, including custom JavaScript libraries, datasources, and themes.
- **Improvements**
	- Streamlined export process with artifact-based services.
- **Documentation**
	- Updated import statements and annotations for clarity.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/exports/internal/partial/PartialExportServiceCEImpl.java [hunk 1]
```diff
@@ -70,6 +70,7 @@ public Mono<ApplicationJson> getPartialExportResources(
         final MappedExportableResourcesDTO mappedResourcesDTO = new MappedExportableResourcesDTO();
         final ExportingMetaDTO exportingMetaDTO = new ExportingMetaDTO();
 
+        exportingMetaDTO.setArtifactType(FieldName.APPLICATION);
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/exports/internal/partial/PartialExportServiceCEImpl.java:73]: It's being used further down

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 2]
```diff
@@ -68,35 +68,35 @@
...
+    public static final String FORM_WIDGET = "FORM_WIDGET";
+    public static final String JSON_FORM_WIDGET = "JSON_FORM_WIDGET";
+    public static final String DROP_DOWN_WIDGET = "DROP_DOWN_WIDGET";
+    public static final String OPTIONS = "options";
+    public static final String DEFAULT_OPTION = "defaultOptionValue";
+    public static final String PRIMARY_COLUMNS = "primaryColumns";
+    public static final String MONGO_ESCAPE_ID = "appsmith_mongo_escape_id";
+    public static final String MONGO_ESCAPE_CLASS = "appsmith_mongo_escape_class";
+    public static final String MONGO_UNESCAPED_ID = "_id";
+    public static final String MONGO_UNESCAPED_CLASS = "_class";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:98]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 3]
```diff
@@ -68,35 +68,35 @@
...
+    public static final String CANVAS_WIDGET = "CANVAS_WIDGET";
+    public static final String FORM_WIDGET = "FORM_WIDGET";
+    public static final String JSON_FORM_WIDGET = "JSON_FORM_WIDGET";
+    public static final String DROP_DOWN_WIDGET = "DROP_DOWN_WIDGET";
+    public static final String OPTIONS = "options";
+    public static final String DEFAULT_OPTION = "defaultOptionValue";
+    public static final String PRIMARY_COLUMNS = "primaryColumns";
+    public static final String MONGO_ESCAPE_ID = "appsmith_mongo_escape_id";
+    public static final String MONGO_ESCAPE_CLASS = "appsmith_mongo_escape_class";
+    public static final String MONGO_UNESCAPED_ID = "_id";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:97]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 4]
```diff
@@ -18,36 +18,36 @@
...
+    public static final String SOURCE_APPLICATION_ID = "sourceApplicationId";
     public static final String DEFAULT_RESOURCES = "defaultResources";
-    public static String PLUGIN_ID = "pluginId";
-    public static String DATASOURCE = "datasource";
-    public static String CONFIG = "config";
-    public static String PLUGIN = "plugin";
-    public static String DEFAULT_PAGE_NAME = "Page1";
-    public static String TYPE = "type";
+    public static final String PLUGIN_ID = "pluginId";
+    public static final String DATASOURCE = "datasource";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:30]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 5]
```diff
@@ -18,36 +18,36 @@
...
+    public static final String CONFIG = "config";
+    public static final String PLUGIN = "plugin";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:32]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 6]
```diff
@@ -18,36 +18,36 @@
...
-    public static String USER = "user";
-    public static String CATEGORY = "category";
-    public static String PAGE = "page";
-    public static String PAGES = "pages";
-    public static String SIZE = "size";
-    public static String ROLE = "role";
-    public static String PROFICIENCY = "proficiency";
-    public static String DEFAULT_WIDGET_NAME = "MainContainer";
-    public static String DEFAULT_PAGE_LAYOUT = "{\n" + "        \"widgetName\": \"MainContainer\",\n"
+    public static final String WIDGET_NAME = "widgetName";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:36]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 7]
```diff
@@ -18,36 +18,36 @@
...
+    public static final String DYNAMIC_BINDINGS = "dynamicBindings";
+    public static final String DYNAMIC_BINDING_PATH_LIST = "dynamicBindingPathList";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:38]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 8]
```diff
@@ -18,36 +18,36 @@
...
+    public static final String KEY = "key";
+    public static final String CHILDREN = "children";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:40]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 9]
```diff
@@ -18,36 +18,36 @@
...
+    public static final String ORIGIN = "origin";
+    public static final String USER = "user";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:42]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 10]
```diff
@@ -18,36 +18,36 @@
...
+    public static final String CATEGORY = "category";
+    public static final String PAGE = "page";
+    public static final String PAGES = "pages";
+    public static final String SIZE = "size";
+    public static final String ROLE = "role";
+    public static final String PROFICIENCY = "proficiency";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:48]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 11]
```diff
@@ -18,36 +18,36 @@
...
+    public static final String DYNAMIC_BINDING_PATH_LIST = "dynamicBindingPathList";
+    public static final String KEY = "key";
+    public static final String CHILDREN = "children";
+    public static final String ORIGIN = "origin";
+    public static final String USER = "user";
+    public static final String CATEGORY = "category";
+    public static final String PAGE = "page";
+    public static final String PAGES = "pages";
+    public static final String SIZE = "size";
+    public static final String ROLE = "role";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:47]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 12]
```diff
@@ -18,36 +18,36 @@
...
+    public static final String DYNAMIC_BINDINGS = "dynamicBindings";
+    public static final String DYNAMIC_BINDING_PATH_LIST = "dynamicBindingPathList";
+    public static final String KEY = "key";
+    public static final String CHILDREN = "children";
+    public static final String ORIGIN = "origin";
+    public static final String USER = "user";
+    public static final String CATEGORY = "category";
+    public static final String PAGE = "page";
+    public static final String PAGES = "pages";
+    public static final String SIZE = "size";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:46]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 13]
```diff
@@ -18,36 +18,36 @@
...
+    public static final String WIDGET_NAME = "widgetName";
+    public static final String DYNAMIC_BINDINGS = "dynamicBindings";
+    public static final String DYNAMIC_BINDING_PATH_LIST = "dynamicBindingPathList";
+    public static final String KEY = "key";
+    public static final String CHILDREN = "children";
+    public static final String ORIGIN = "origin";
+    public static final String USER = "user";
+    public static final String CATEGORY = "category";
+    public static final String PAGE = "page";
+    public static final String PAGES = "pages";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:45]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 14]
```diff
@@ -18,36 +18,36 @@
...
-    public static String DEFAULT_WIDGET_NAME = "MainContainer";
-    public static String DEFAULT_PAGE_LAYOUT = "{\n" + "        \"widgetName\": \"MainContainer\",\n"
+    public static final String WIDGET_NAME = "widgetName";
+    public static final String DYNAMIC_BINDINGS = "dynamicBindings";
+    public static final String DYNAMIC_BINDING_PATH_LIST = "dynamicBindingPathList";
+    public static final String KEY = "key";
+    public static final String CHILDREN = "children";
+    public static final String ORIGIN = "origin";
+    public static final String USER = "user";
+    public static final String CATEGORY = "category";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:43]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 15]
```diff
@@ -18,36 +18,36 @@
...
-    public static String CATEGORY = "category";
-    public static String PAGE = "page";
-    public static String PAGES = "pages";
-    public static String SIZE = "size";
-    public static String ROLE = "role";
-    public static String PROFICIENCY = "proficiency";
-    public static String DEFAULT_WIDGET_NAME = "MainContainer";
-    public static String DEFAULT_PAGE_LAYOUT = "{\n" + "        \"widgetName\": \"MainContainer\",\n"
+    public static final String WIDGET_NAME = "widgetName";
+    public static final String DYNAMIC_BINDINGS = "dynamicBindings";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:37]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 16]
```diff
@@ -18,36 +18,36 @@
...
-    public static String CONFIG = "config";
-    public static String PLUGIN = "plugin";
-    public static String DEFAULT_PAGE_NAME = "Page1";
-    public static String TYPE = "type";
+    public static final String PLUGIN_ID = "pluginId";
+    public static final String DATASOURCE = "datasource";
+    public static final String CONFIG = "config";
+    public static final String PLUGIN = "plugin";
+    public static final String DEFAULT_PAGE_NAME = "Page1";
+    public static final String TYPE = "type";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:34]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 17]
```diff
@@ -18,36 +18,36 @@
...
-    public static String DATASOURCE = "datasource";
-    public static String CONFIG = "config";
-    public static String PLUGIN = "plugin";
-    public static String DEFAULT_PAGE_NAME = "Page1";
-    public static String TYPE = "type";
+    public static final String PLUGIN_ID = "pluginId";
+    public static final String DATASOURCE = "datasource";
+    public static final String CONFIG = "config";
+    public static final String PLUGIN = "plugin";
+    public static final String DEFAULT_PAGE_NAME = "Page1";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:33]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 18]
```diff
@@ -18,36 +18,36 @@
...
     public static final String DEFAULT_RESOURCES = "defaultResources";
-    public static String PLUGIN_ID = "pluginId";
-    public static String DATASOURCE = "datasource";
-    public static String CONFIG = "config";
-    public static String PLUGIN = "plugin";
-    public static String DEFAULT_PAGE_NAME = "Page1";
-    public static String TYPE = "type";
+    public static final String PLUGIN_ID = "pluginId";
+    public static final String DATASOURCE = "datasource";
+    public static final String CONFIG = "config";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:31]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 19]
```diff
@@ -18,36 +18,36 @@
...
     public static final String COLLECTION_ID = "collectionId";
     public static final String ACTION_ID = "actionId";
-    public static String WORKSPACE = "workspace";
-    public static String ID = "id";
+    public static final String WORKSPACE = "workspace";
+    public static final String ID = "id";
     public static final String NAME = "name";
-    public static String PAGE_ID = "pageId";
-    public static String LAYOUT_ID = "layoutId";
+    public static final String PAGE_ID = "pageId";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:24]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 20]
```diff
@@ -18,36 +18,36 @@
     public static final String PACKAGE_NAME = "packageName";
     public static final String COLLECTION_ID = "collectionId";
     public static final String ACTION_ID = "actionId";
-    public static String WORKSPACE = "workspace";
-    public static String ID = "id";
+    public static final String WORKSPACE = "workspace";
+    public static final String ID = "id";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:22]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 21]
```diff
@@ -18,36 +18,36 @@ public class FieldNameCE {
     public static final String PACKAGE_NAME = "packageName";
     public static final String COLLECTION_ID = "collectionId";
     public static final String ACTION_ID = "actionId";
-    public static String WORKSPACE = "workspace";
-    public static String ID = "id";
+    public static final String WORKSPACE = "workspace";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:21]: Not a new varNot a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 22]
```diff
@@ -18,36 +18,36 @@
...
-    public static String SOURCE_APPLICATION_ID = "sourceApplicationId";
+    public static final String SOURCE_APPLICATION_ID = "sourceApplicationId";
     public static final String DEFAULT_RESOURCES = "defaultResources";
-    public static String PLUGIN_ID = "pluginId";
-    public static String DATASOURCE = "datasource";
-    public static String CONFIG = "config";
-    public static String PLUGIN = "plugin";
-    public static String DEFAULT_PAGE_NAME = "Page1";
-    public static String TYPE = "type";
+    public static final String PLUGIN_ID = "pluginId";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:29]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 23]
```diff
@@ -18,36 +18,36 @@
...
-    public static String ROLE = "role";
-    public static String PROFICIENCY = "proficiency";
-    public static String DEFAULT_WIDGET_NAME = "MainContainer";
-    public static String DEFAULT_PAGE_LAYOUT = "{\n" + "        \"widgetName\": \"MainContainer\",\n"
+    public static final String WIDGET_NAME = "widgetName";
+    public static final String DYNAMIC_BINDINGS = "dynamicBindings";
+    public static final String DYNAMIC_BINDING_PATH_LIST = "dynamicBindingPathList";
+    public static final String KEY = "key";
+    public static final String CHILDREN = "children";
+    public static final String ORIGIN = "origin";
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:41]: Not a new var

File: app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java [hunk 24]
```diff
@@ -126,7 +126,7 @@
             + "editing applications, inviting other users to the workspace and exporting applications "
             + "from the workspace";
     public static final String DEVELOPER = "Developer";
-    public static String WORKSPACE_DEVELOPER_DESCRIPTION =
+    public static final String WORKSPACE_DEVELOPER_DESCRIPTION =
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/constants/ce/FieldNameCE.java:129]: Not a new var

**Review by sondermanish — APPROVED** [review]

**Pull request merged**